airqo-platform/AirQo-api
# Role based access to platform endpoints

# Role based access to platform endpoints

**_WHAT DOES THIS PR DO?_**
This PR introduces role based access control to the auth service

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
[PLAT-293](https://airqoteam.atlassian.net/browse/PLAT-293?atlOrigin=eyJpIjoiMjM3NjJhNWU3ZjM0NGZiMGJiNDczNjc3ZGYzMmY3ZmQiLCJwIjoiaiJ9)

**_WHAT IS THE LINK TO THE PR BRANCH_**
https://github.com/airqo-platform/AirQo-api/tree/ft-role-based-access

**_HOW DO I TEST OUT THIS PR?_**
`cd AirQo-api/auth-service`
`npm install`

> for Windows

`npm run stage-pc`

> for Linux

`npm run stage-mac`

And then after, using the Login endpoint, 

- [x] _Login_ using an account whose role is "user" 

- [x] Then try to access the _Get All Users_ endpoint. Remember to use the JWT Token in your HEADERS.

Repeat the process using an account whose role is "admin"

You should be denied access to view users if your role is "user" and you should be allowed to view them if you are "_admin_" or "_super_"

The ENVs and config files are here: https://drive.google.com/drive/folders/1rW4p1l4aD4zLuigQXOhJN7d8tF3M7n9J
Config files are under config sub directory

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

1. Login [POST]: http://localhost:3000/api/v1/users/loginUser?tenant=airqo
2. Get All Users [POST]: http://localhost:3000/api/v1/users?tenant=airqo

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
N/A

**_ARE THERE ANY RELATED PRs?_**
N/A
